src/TimeSteppers/clock.jl

```diff
@@ -6,30 +6,33 @@ import Base: show
 import Oceananigans.Units: Time
 
 """
-    mutable struct Clock{T<:Number}
+    mutable struct Clock{T, FT}
 
-Keeps track of the current `time`, `iteration` number, and time-stepping `stage`.
+Keeps track of the current `time`, `last_Δt`, `iteration` number, and time-stepping `stage`.
 The `stage` is updated only for multi-stage time-stepping methods. The `time::T` is
 either a number or a `DateTime` object.
 """
-mutable struct Clock{T}
-         time :: T
+mutable struct Clock{TT, DT}
+         time :: TT
+      last_Δt :: DT
     iteration :: Int
         stage :: Int
 end
 
 """
-    Clock(; time, iteration=0, stage=1)
+    Clock(; time, last_Δt = Inf, iteration=0, stage=1)
 
 Returns a `Clock` object. By default, `Clock` is initialized to the zeroth `iteration`
-and first time step `stage`.
+and first time step `stage` with `last_Δt`.
 """
-Clock(; time::T, iteration=0, stage=1) where T = Clock{T}(time, iteration, stage)
+Clock(; time::TT, last_Δt::DT=Inf, iteration=0, stage=1) where {TT, DT} = Clock{TT, DT}(time, last_Δt, iteration, stage)
+Clock{TT}(; time, last_Δt=Inf, iteration=0, stage=1) where TT = Clock{TT, TT}(time, last_Δt, iteration, stage)
+Clock{DT}(; time::AbstractTime, last_Δt=Inf, iteration=0, stage=1) where DT = Clock{AbstractTime, DT}(time, last_Δt, iteration, stage)

```

Comment:
```suggestion
# TODO: when supporting DateTime, this function will have to be extended
time_step_type(TT) = TT

function Clock{TT}(; time, last_Δt=Inf, iteration=0, stage=1) where TT
    DT = time_step_type(TT)
    last_Δt = convert(DT, last_Δt)
    return Clock{TT, DT}(time, last_Δt, iteration, stage)
end
```

And get rid of the `AbstractTime` function since I don't think it's what we want (also we don't properly support `AbstractTime` anyways)